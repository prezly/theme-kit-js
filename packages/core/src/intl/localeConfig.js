const { Locale } = require('@prezly/theme-kit-intl');

/**
 * This list is pulled from the main Prezly application. The underscores are replaced with dashes.
 * Together with permutation performed lower in the code,
 * the exported list represents all possible locale codes that a theme application might accept.
 */

const { isNumberCode } = require('./languages');
/**
 * @type {Locale[]}
 */
const locales = require('./localeList').map((code) => Locale.from(code));

/**
 * All permutations for the shorter locale codes supported by Prezly (e.g. short region codes and neutral language codes)
 * @type {string[]}
 */
const permutations = Array.from(
    new Set([
        ...locales.map((locale) => locale.slug),
        ...locales.map((locale) => locale.lang),
        ...locales
            .map((locale) => locale.region?.toLowerCase())
            .filter(Boolean)
            // Remove number-only codes from possible permutations (like `419` for `es-419`)
            .filter((code) => !isNumberCode(code))
            // Remove codes that are too long to be valid standalone codes (like `hant` for `zh-hant`)
            .filter((code) => code.length <= 3),
    ]),
).sort();

module.exports = permutations;
