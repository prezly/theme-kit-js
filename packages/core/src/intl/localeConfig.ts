/**
 * This list is pulled from the main Prezly application. The underscores are replaced with dashes.
 * Together with permutation performed lower in the code,
 * the exported list represents all of the possible locale codes that a theme application might accept.
 */
import supportedLocales from './localeList.js';

const lowercaseLocales = supportedLocales.map((l) => l.toLowerCase());

// Get all permutations for the shorter locale codes supported by Prezly (e.g. short region codes and neutral language codes)
const lowercaseLocalPermutations = Array.from(
    new Set([
        ...lowercaseLocales,
        ...lowercaseLocales.map((code) => code.split('-')[0]),
        ...lowercaseLocales
            .map((code) => code.split('-')[1])
            .filter(Boolean)
            // Remove number-only codes from possible permutations (like `419` for `es-419`)
            .filter((code) => Number.isNaN(Number(code))),
    ]),
).sort();

// eslint-disable-next-line import/no-default-export
export default lowercaseLocalPermutations;
