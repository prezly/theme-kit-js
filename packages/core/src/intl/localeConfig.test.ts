import localeConfig from './localeConfig';

describe('localeConfig', () => {
    it('should have all possible permutations of locale, language and region codes', () => {
        expect(localeConfig).toMatchSnapshot();
    });
});
