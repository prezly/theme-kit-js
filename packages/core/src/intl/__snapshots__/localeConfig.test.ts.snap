// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`localeConfig should have all possible permutations of locale, language and region codes 1`] = `
[
  "ae",
  "af",
  "af-za",
  "ar",
  "ar-ae",
  "ar-bh",
  "ar-qa",
  "ar-sa",
  "at",
  "au",
  "az",
  "bd",
  "be",
  "bg",
  "bg-bg",
  "bh",
  "br",
  "ca",
  "ch",
  "cn",
  "cs",
  "cs-cz",
  "cz",
  "da",
  "da-dk",
  "de",
  "de-at",
  "de-be",
  "de-ch",
  "de-de",
  "dk",
  "ee",
  "el",
  "el-gr",
  "en",
  "en-ae",
  "en-au",
  "en-bd",
  "en-be",
  "en-bh",
  "en-ca",
  "en-cn",
  "en-de",
  "en-dk",
  "en-gb",
  "en-hk",
  "en-id",
  "en-ie",
  "en-il",
  "en-in",
  "en-jm",
  "en-jp",
  "en-kh",
  "en-kr",
  "en-lk",
  "en-mm",
  "en-mv",
  "en-my",
  "en-np",
  "en-nz",
  "en-ph",
  "en-pk",
  "en-qa",
  "en-sa",
  "en-sg",
  "en-th",
  "en-tw",
  "en-us",
  "en-vn",
  "en-za",
  "es",
  "es-419",
  "es-es",
  "es-mx",
  "et",
  "et-ee",
  "fi",
  "fi-fi",
  "fil",
  "fr",
  "fr-be",
  "fr-ca",
  "fr-ch",
  "fr-fr",
  "fr-lu",
  "ga",
  "ga-ie",
  "gb",
  "gr",
  "he",
  "he-il",
  "hi",
  "hi-in",
  "hk",
  "hr",
  "hr-hr",
  "hu",
  "hu-hu",
  "id",
  "id-id",
  "ie",
  "il",
  "in",
  "it",
  "it-it",
  "ja",
  "ja-jp",
  "jm",
  "jp",
  "kh",
  "kk",
  "ko",
  "ko-kr",
  "kr",
  "ky",
  "lk",
  "lt",
  "lt-lt",
  "lu",
  "lv",
  "lv-lv",
  "mm",
  "ms",
  "ms-my",
  "mt",
  "mt-mt",
  "mv",
  "mx",
  "my",
  "nb",
  "nb-no",
  "nl",
  "nl-be",
  "nl-nl",
  "no",
  "np",
  "nz",
  "ph",
  "pk",
  "pl",
  "pl-pl",
  "pt",
  "pt-br",
  "pt-pt",
  "qa",
  "ro",
  "ro-ro",
  "rs",
  "ru",
  "ru-ru",
  "sa",
  "se",
  "sg",
  "si",
  "sk",
  "sk-sk",
  "sl",
  "sl-si",
  "sr",
  "sr-rs",
  "sv",
  "sv-se",
  "sw",
  "th",
  "th-th",
  "tr",
  "tr-tr",
  "tw",
  "ua",
  "uk",
  "uk-ua",
  "ur",
  "ur-pk",
  "us",
  "uz",
  "vi",
  "vi-vn",
  "vn",
  "za",
  "zh",
  "zh-cn",
  "zh-hant",
  "zh-hk",
  "zh-tw",
]
`;
